File: vacgom-persistence/src/main/java/kr/co/vacgom/persistence/vaccination/infrastructure/VaccinationRepositoryAdapter.java
package kr.co.vacgom.persistence.vaccination.infrastructure;

import kr.co.vacgom.persistence.vaccination.entity.VaccinationEntity;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Repository;

import java.util.List;

@Repository
@RequiredArgsConstructor
class VaccinationRepositoryAdapter implements VaccinationRepository {

    private final VaccinationQueryDslRepository vaccinationQueryDslRepository;
    private final VaccinationJpaRepository vaccinationJpaRepository;

    @Override
    public VaccinationEntity save(VaccinationEntity vaccination) {
        return null;
    }

    @Override
    public List<VaccinationEntity> saveAll(List<VaccinationEntity> vaccinations) {
        return List.of();
    }
}

File: src/main/kotlin/kr/co/vacgom/api/vaccine/repository/VaccinationRepositoryAdapter.kt
package kr.co.vacgom.api.vaccine.repository

import kr.co.vacgom.api.vaccine.domain.Vaccination
import org.springframework.stereotype.Repository

@Repository
class VaccinationRepositoryAdapter(
    private val vaccinationJpaRepository: VaccinationJpaRepository
) : VaccinationRepository {

    override fun save(vaccination: Vaccination): Vaccination {
        return vaccinationJpaRepository.save(vaccination)
    }
}
